src/lib/updating-element.ts

```diff
@@ -520,11 +527,18 @@ export abstract class UpdatingElement extends HTMLElement {
 
   /**
    * Finishes updating the element. This method does nothing by default and
-   * should be implemented to perform post update tasks on element DOM. This
-   * async function can await other Promises to defer the resolution of the
-   * `invalidate` and `updateComplete` Proimses.
+   * can be implemented to perform post update tasks on element DOM.
    * * @param _changedProperties changed properties with old values
+   * * @returns {Promise} Optionally can return a promise that blocks
+   * resolution of the `invalidate` and updateComplete` promise.
    */
-  protected finishUpdate?(_changedProperties: PropertyValues): void;
+  protected finishUpdate?(_changedProperties: PropertyValues): void|Promise<any>;
 
+  /**
+   * Called with the element is first updated. This method does nothing by
+   * default and can be implemented to perform post first update tasks on
+   * element DOM. Any tasks which should synchronous with dynamic updates
+   * should be implemented in `finishUpdate`.
+   */
+  protected firstUpdated?(): void;
```

Comment:
firstUpdated sounds like a boolean to me. 

Could be be an event instead? or maybe a promise?

```
await this.firstUpdateComplete();
...
```

Reply:
Current LitElement already has 

```
renderComplete: Returns a promise which resolves after the element next renders.
```

So it could just be a specialized version of that.

Also depending on where you call this, you could listen to the first render. If I for instance do ```renderComplete().then(...)``` inside the ctor.